vault/service/init.sls

```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+# vim: ft=sls syntax=yaml softtabstop=2 tabstop=2 shiftwidth=2 expandtab autoindent
+
+{% from "vault/map.jinja" import vault with context %}
+
+vault-service-init-file-managed:
+  file.managed:
+    - name: {{ vault.service.path }}
+    - source: {{ vault.service.source }}
+    - template: jinja
+    - watch_in:
+      - service: vault
+
+vault-service-init-service-running:
+  service.running:
+    - name: vault
+    - enable: true
+    - watch:
+      - vault-package-install-archive-extracted
+      - vault-service-init-file-managed
+      - vault-config-init-file-serialize
```

Comment:
Redundant use of both `watch` and `watch_in`.  Easier to keep the `watch`, since `vault-package-install-archive-extracted` is missing the `watch_in`.  If selecting the `watch_in` instead, then the missing `watch_in` needs to be added.

Reply:
If you look closely you can see that the `watch_in` refers to a non-existing id. I missed this while cleaning up.

Reply:
@rbjorklin Note, these are missing the state type (`archive`, `file` & `file` respectively).  So should be:

```sls
    - watch:
      - archive: vault-package-install-archive-extracted
      - file: vault-service-init-file-managed
      - file: vault-config-config-file-serialize
```

Reply:
Not strictly necessary any more: https://docs.saltstack.com/en/latest/ref/states/requisites.html#omitting-state-module-in-requisites

Reply:
This is generally kept explicit across formulas.

Reply:
I don't have a strong opinion either way, I just kept it the way it was in this formula. I'll let @myoung34 have the final word here.

Reply:
I usually keep explicit